shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/keygen/UUIDKeyGenerateAlgorithm.java

```diff
@@ -31,8 +33,8 @@ public void init() {
     }
     
     @Override
-    public synchronized Comparable<?> generateKey() {
-        return UUID.randomUUID().toString().replaceAll("-", "");
+    public Comparable<?> generateKey() {
+        return StringUtils.replace(new UUID(ThreadLocalRandom.current().nextLong(), ThreadLocalRandom.current().nextLong()).toString(), "-", "");
```

Comment:
Why use the `StringUtils`? What is the different between `StringUtils` and `String.replace`?

Reply:
> 

String.replace uses regular expressions.
commons-lang3 StringUtils.replace uses string Interception, restructuring.
Regular expressions are inefficient.

Reply:
> > 
> 
> String.replace uses regular expressions.
> commons-lang3 StringUtils.replace uses string Interception, restructuring.
> Regular expressions are inefficient.
> 
> 

String.replaceAll uses regex indeed, but String replace may not. Could you check it again?

Reply:
jdk1.8 java.lang.String   
```
public String replace(CharSequence target, CharSequence replacement) {
        return Pattern.compile(target.toString(), Pattern.LITERAL).matcher(this).replaceAll(Matcher.quoteReplacement(replacement.toString()));
}
```

Reply:
> jdk1.8 java.lang.String
> 
> ```
> public String replace(CharSequence target, CharSequence replacement) {
>         return Pattern.compile(target.toString(), Pattern.LITERAL).matcher(this).replaceAll(Matcher.quoteReplacement(replacement.toString()));
> }
> ```

I see. The `String.replace` is improved since Java 9.

Reply:
Oh, I've learned a thing.